parquet-thrift/src/main/java/parquet/thrift/ThriftRecordConverter.java

```diff
@@ -843,7 +843,9 @@ public T getCurrentRecord() {
       rootEvents.clear();
       return thriftReader.readOneRecord(protocol);
     } catch (TException e) {
-      throw new ParquetDecodingException("Could not read thrift object from protocol", e);
+      protocol.clear();
+      rootEvents.clear();
```

Comment:
We can probably refactor this, but the reason is rootEvents are events that we plan to replay through the protocol. In the case of skipping a record, clearing this is enough. In the case of an exception, we have already pushed rootEvents into protocol, which has now only partially consumed those events. So both need to be cleared. I think this is correct, and we can refactor separately.